### ISSUE TITLE
[Bug]: Not reading properties yet loads page fine

### ISSUE THREAD

**Issue by dcalitri** [FreeTubeApp/FreeTube#4811]
### Developer/Maintainer edit
Before you leave a comment, please read https://github.com/FreeTubeApp/FreeTube/issues/4811#issuecomment-2019474416.

### Guidelines

- [X] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [X] I have encountered this bug in the [official downloads of FreeTube](https://github.com/FreeTubeApp/FreeTube#official-downloads).
- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the bug report I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.
- [X] This issue contains only one bug.

### Describe the bug

Whenever I open the app or refresh the subscription page, I keep getting multiple instances of this error even though I'm connected to wifi.

### Expected Behavior

I don't get this error when I open the app.

### Issue Labels

API issue

### FreeTube Version

v0.19.2 Beta

### Operating System Version

Windows 11

### Installation Method

.exe

### Primary API used

Local API

### Last Known Working FreeTube Version (If Any)

v0.19.2 Beta

### Additional Information

![Freetube Error](https://github.com/FreeTubeApp/FreeTube/assets/71895943/30ded470-adf2-41f0-9b6a-3be545cb13b1)

### Nightly Build

- [ ] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

**Comment by thingsiplay** [comment]
I want to confirm this issue on my Linux PC wired internet connection.

As an alternative one could fetch from RSS (Settings > Subscription Settings > Fetch Feeds from RSS) if this bothers until a solution is found. Or use a different API (Settings > General Settings > Preferred API Backend = "Indivious API"), but that is much slower on my machine.

**Comment by bmgxyz** [comment]
Just tested on [latest nightly](https://github.com/FreeTubeApp/FreeTube/actions/runs/8429199872) and confirmed that the issue exists there too.

Also, latest nightly reports itself as version 0.19.1 and suggests an upgrade to 0.19.2. Am I doing something wrong, or is this also an issue?

**Comment by bmgxyz** [comment]
Doesn't appear in my local dev environment, even with a channel added to the subscription list. Is there a way to import my existing user state into the development version? I poked around, but it's not obvious to me how to accomplish that.

**Comment by absidue** [comment]
FreeTube already automatically falls back to RSS for the channels that error, so you can ignore the error for now, as your subscriptions will still work just fine because of the automatic RSS fallback.

> Just tested on latest nightly and confirmed that the issue exists there too.

Yes that's what happens when YouTube changes stuff, it breaks everywhere, although if you had waited a few days for the pull request to be merged, that I had already opened before this issue was opened, you wouldn't be seeing the error anymore.

> Also, latest nightly reports itself as version 0.19.1 and suggests an upgrade to 0.19.2. Am I doing something wrong, or is this also an issue?

That is entirely intentional the nightlies have all the development changes since 0.19.1 was released, 0.19.2 cherry-picked a few specific changes/fixes from the development branch, so that people could continue to use FreeTube until 0.20.0 is released. We'll bump the version on the development branch to 0.20.0 when 0.20.0 is released, because then all development branch changes, will have been released, if we had bumped it for 0.19.2 it would have been a lie. For the nightlies you can ignore the version entirely, what matters there is the build number, if it's bigger it's more recent one, if it's lower it's an older one.

> Doesn't appear in my local dev environment, even with a channel added to the subscription list.

It's part of a YouTube A/B test, which is why the error only appears for a few random channels for every refresh, you probably don't have enough subscriptions in your dev environment to have a high enough chance for one of the channels to error.

> Is there a way to import my existing user state into the development version? I poked around, but it's not obvious to me how to accomplish that.

The export and import buttons in the Data Settings section...

**Comment by z0x0936** [comment]
TypeError: Cannot read properties of undefined (reading 'text') - I get several of these and seems related to OP. Just started seeing this error as of yesterday (if memory serves correctly)

**Comment by absidue** [comment]
TL;DR:
- The error is caused by a YouTube A/B test (means YouTube is testing new stuff, that FreeTube doesn't support yet), it it will only happen for a few random channels for each refresh.
- You can ignore the error, yes it might be annoying, but because FreeTube automatically falls back to RSS for the channels that error, your subscriptions will continue to work just fine.
- A pull request adding support for the YouTube changes was already opened hours before this issue was opened, so yes a change to fix the error is coming, but we won't be rushing out a release specifically for that because the RSS fallback is working completely fine.

**Comment by bmgxyz** [comment]
>A pull request adding support for the YouTube changes was already opened

Nice, thanks. #4804

**Comment by absidue** [comment]
Closing as this was fixed in https://github.com/FreeTubeApp/FreeTube/pull/4804 which was released in 0.20.0. We left this issue open for longer than usual, to avoid people creating loads of duplicate ones until it was released.
